File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int n;
    do
    {
        // prompt
        n = get_int("Enter a positive integer less of 23 or less: ");
    }
    // repeat if incorrect input
    while (n < 0 || n > 23);
    // forloop for each row
    for (int i = 0; i < n; i++)
    {
        // calculate leading space and hash numbers
        int space = n - (i + 1);
        int hash = n - space;

        // print leading spaces
        for (int j = 0; j < space; j++)
        {
            printf(" ");
        }
        // print first set of hashes
        for (int k = 0; k < hash; k++)
        {
            printf("#");
        }
        // print standard 2-space separator
        printf("  ");
        // print second set of hashes
        for (int k = 0; k < hash; k++)
        {
            printf("#");
        }
        // new line to prep for next row forloop
        printf("\n");
    }
}

File: pset6/mario/mario.py
# prints pyramid based on user input
from cs50 import get_int

while True:
    n = get_int("Enter a positive integer of 23 or less: ")
    if n >= 0 and n <= 23:
        break
for i in range(n):
    # calculate leading spaces and hashes
    spaces = n - (i + 1)
    hashes = n - spaces

    # print leading spaces
    print(" " * spaces, end="")

    # print first set of hashes
    print("#" * hashes, end="")

    # print 2-space separator
    print(" " * 2, end="")

    # print second set of hashes
    print("#" * hashes, end="")

    # print new line
    print()